src/test/test_confparse.c

```diff
@@ -906,11 +906,8 @@ test_confparse_unitparse(void *args)
   tt_assert(ok);
 
   /* u64 overflow */
-  /* XXXX our implementation does not currently detect this. See bug 30920. */
-  /*
   tt_u64_op(config_parse_memunit("20000000 TB", &ok), OP_EQ, 0);
```

Comment:
We'll also need tests for the floating-point case.
It would be great to have a test that fails the float comparison, and then one that passes the float comparison, but fails the int comparison.

Reply:
I've never dealt with float operations in such a context so I had to read about it a bit before. I'm pasting here what I found (by testing in a separate c file) to be a case for each failing comparison before submitting.

In this case the double comparison will fail
```
 // Fails double comparison
 tt_u64_op(config_parse_memunit("15000000.5 TB", &ok), OP_EQ, 0);
 tt_assert(!ok);
```

This case is a valid float number, (`INT64_MAX` - 0.5) but its integer representation is greater than INT64_MAX.
```
 // Fails int comparison
 tt_u64_op(config_parse_memunit("9223372036854775806.5 B", &ok), OP_EQ, 0);
 tt_assert(!ok);
```